pkg/gc/finalizer.go

```diff
@@ -140,38 +141,55 @@ func (f *subscriptionFinalizer) checkGeneration(ctx context.Context, typ core_mo
 	f.removeDeletedInsights(insights, tenantId)
 
 	for _, item := range insights.GetItems() {
-		log := finalizerLog.WithValues("type", typ, "name", item.GetMeta().GetName(), "mesh", item.GetMeta().GetMesh())
+		logger := finalizerLog.WithValues("type", typ, "name", item.GetMeta().GetName(), "mesh", item.GetMeta().GetMesh())
+		logger = log.AddFieldsFromCtx(logger, ctx, f.extensions)
```

Comment:
Why do we need these extensions everywhere in the logger? I've used them as well and it's super annoying and confusing to folks...

Feels like this may be one of the rare places where a global might be easier to use...

Reply:
Hoo https://github.com/kumahq/kuma/pull/8184 support of slog in Logger. Wondering if InfoWithContext/ErrorWithContext can make all this a lot easier

Reply:
@lahabana should we reexamine AddFieldsFromCtx in a differnet issue?

Reply:
Ho yes of course!

Reply:
https://github.com/kumahq/kuma/issues/8216